File: poc-kronos-frontend/src/model/funcionario/Ponto.js
class Ponto {
  constructor(dados) {
    this.id = dados ? dados.id : null;
    this.tarefaOnline = dados ? dados.tarefaOnline : null;
    this.horaInicialRegistro = dados ? dados.horaInicialRegistro : null;
    this.horaFinalRegistro = dados ? dados.horaFinalRegistro : null;
    this.horaInicialInformado = dados ? dados.horaInicialInformado : null;
    this.horaFinalInformado = dados ? dados.horaFinalInformado : null;
    this.data = dados ? dados.data : null;
    this.idTipoPeriodoPonto = dados ? dados.idTipoPeriodoPonto : null;
    this.nomeTipoPeriodoPonto = dados ? dados.nomeTipoPeriodoPonto : null;
    this.idConfiguracaoPonto = dados ? dados.idConfiguracaoPonto : null;
    this.idFuncionarioLiberacao = dados ? dados.idFuncionarioLiberacao : null;
    this.dataLiberacao = dados ? dados.dataLiberacao : null;
    this.dataHomologacao = dados ? dados.dataHomologacao : null;
    this.idFuncionarioHomologacao = dados
      ? dados.idFuncionarioHomologacao
      : null;
    this.descricao = dados ? dados.descricao : null;
    this.horaInicialPrevista = dados ? dados.horaInicialPrevista : null;
    this.horaFinalPrevista = dados ? dados.horaFinalPrevista : null;
    this.toleranciaEntrada = dados ? dados.toleranciaEntrada : null;
    this.toleranciaSaida = dados ? dados.toleranciaSaida : null;
    this.dataHoje = dados ? dados.dataHoje : null;
    this.pendente = false;
    this.idFuncionario = null;
    this.ano = null;
    this.mes = null;
    this.diaSemana = dados ? dados.diaSemana : null;
    this.funcionarioLiberacao = dados ? dados.funcionarioLiberacao : null;
    this.funcionarioHomologacao = dados ? dados.funcionarioHomologacao : null;
    this.idsPontos = [];
  }

  static validations(required, maxLength, dataFimMenor, horaFinalMenor) {
    return {
      descricao: { maxLength: maxLength(200) },
      idTipoPeriodoPonto: { required },
      horaFinalInformado: {
        horaFinalMenor: horaFinalMenor("horaInicialInformado", true)
      }
    };
  }

  static validationMessages() {
    return {
      descricao: {
        maxLength: "Nome só aceita no máximo 200 caracteres"
      },
      idTipoPeriodoPonto: {
        required: "Tipo Período é Obrigatório"
      },
      horaFinalInformado: {
        horaFinalMenor:
          "Hora Final Informado não pode ser menor ou igual a Hora Inicial Informado"
      }
    };
  }
}
export default Ponto;

File: src/main/java/br/com/kronos/backend/aplicacao/funcionario/Ponto.java
package br.com.kronos.backend.aplicacao.funcionario;

import static br.com.kronos.backend.aplicacao.validador.Validacoes.exigirNaoNulo;
import static br.com.kronos.backend.aplicacao.validador.Validacoes.exigirTamanho;

import java.time.LocalDate;
import java.time.LocalTime;

import lombok.Builder;
import lombok.Getter;


@Getter
public class Ponto {
	
	public static final String CACHE_NAME = "pontos";
	public static final String CACHE_NAME_AUSENTES = "ausentes";
	
	private Long id;
	private boolean tarefaOnline;
    private LocalTime horaInicialRegistro;
    private LocalTime horaFinalRegistro;
    private LocalTime horaInicialInformado;
    private LocalTime horaFinalInformado;
    private LocalDate data;
    private Integer idTipoPeriodoPonto;
    private Long idConfiguracaoPonto;
    private Long idFuncionarioLiberacao;
    private LocalDate dataLiberacao;
    private LocalDate dataHomologacao;
    private Long idFuncionarioHomologacao;
    private String descricao;
    
    @Builder
    public Ponto(Long id, boolean tarefaOnline, LocalTime horaInicialRegistro, LocalTime horaFinalRegistro,
			LocalTime horaInicialInformado, LocalTime horaFinalInformado, LocalDate data, Integer idTipoPeriodoPonto,
			Long idConfiguracaoPonto, Long idFuncionarioLiberacao, LocalDate dataLiberacao, LocalDate dataHomologacao,
			Long idFuncionarioHomologacao, String descricao) {
		this.id = id;
		this.tarefaOnline = tarefaOnline;
		this.horaInicialRegistro = horaInicialRegistro;
		this.horaFinalRegistro = horaFinalRegistro;
		this.horaInicialInformado = horaInicialInformado;
		this.horaFinalInformado = horaFinalInformado;
		this.data = exigirNaoNulo(data, "Data");
		this.idTipoPeriodoPonto = exigirNaoNulo(idTipoPeriodoPonto, "Tipo Período PontoDTO");
		this.idConfiguracaoPonto = exigirNaoNulo(idConfiguracaoPonto, "Configuração PontoDTO");
		this.idFuncionarioLiberacao = idFuncionarioLiberacao;
		this.dataLiberacao = dataLiberacao;
		this.dataHomologacao = dataHomologacao;
		this.idFuncionarioHomologacao = idFuncionarioHomologacao;
		this.descricao = exigirTamanho(descricao, "Descrição", 0, 200);
	}

	
}
